compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3301,17 +3302,16 @@ private BType checkInvocationParam(BLangInvocation iExpr) {
         }
 
         List<BType> paramTypes = ((BInvokableType) iExpr.symbol.type).getParameterTypes();
-        Map<String, BVarSymbol> params = ((BInvokableSymbol) iExpr.symbol).params
-                .stream().collect(Collectors.toMap(a -> a.name.getValue(), a -> a));
-        int parameterCount;
-        if (iExpr.symbol.tag == SymTag.VARIABLE) {
-            // Here we assume function pointers can have only required params.
-            // And assume that named params and rest params are not supported.
-            parameterCount = paramTypes.size();
-        } else {
-            parameterCount = ((BInvokableSymbol) iExpr.symbol).params.size();
+        Map<String, BVarSymbol> params = new HashMap<>();
+        for (BVarSymbol a : ((BInvokableSymbol) iExpr.symbol).params) {
+            if (!a.name.equals(Names.EMPTY)) {
+                if (params.put(a.name.getValue(), a) != null) {
+                    throw new IllegalStateException("Duplicate key");
```

Comment:
Shouldn't we add this to dlog.error?

Reply:
This was a mistake when using IDEs feature to automatically convert streams to for loops. :(

This is fixed with https://github.com/ballerina-platform/ballerina-lang/pull/19976/commits/ffa6998300654cdda213da9ca270905b968940d0